### PR TITLE
Minor changes to improve Firefox support

### PR DESCRIPTION
**Pull request by dkanada**
Firefox is really screwed up right now, and I'm not sure exactly why but it might have been a recent update. Can someone try an older version and see if the same issues exist? I added the charset but another ten issues were hiding below that.

### CODE REVIEW

**Comment by telans** [comment]
What exactly is wrong with it for you? Seems fine to me

**Comment by dkanada** [comment]
@telans what version of Firefox and Jellyfin are you using?

**Comment by telans** [comment]
Mozilla Firefox 70.0.1
Jellyfin 10.4.2 (10.4.1 looks the same)

**Comment by dkanada** [comment]
Oh, the issue is on the master branch, not the latest release.

**Comment by dmitrylyzo** [comment]
Firefox 70.0.1 (64-bit)
Works: 957184b81c59e6a0b16a99a62fe1e4958bd58eb2
Black screen: c1b866d9c93ad87f0e1184648eaf46fbed1e8cf5
<details>

```
web socket closed apiclientcore.js:152:21
nulling out web socket apiclientcore.js:152:91
SecurityError: The operation is insecure. appStorage.js:45
Error: Assertion failed bundle.js:532:1602
    a http://192.168.2.18:8096/web/bundle.js?v=10.4.2:532
    _ http://192.168.2.18:8096/web/bundle.js?v=10.4.2:532
    insertBefore http://192.168.2.18:8096/web/bundle.js?v=10.4.2:532
    appendChild http://192.168.2.18:8096/web/bundle.js?v=10.4.2:532
    t http://192.168.2.18:8096/web/bundle.js?v=10.4.2:532
    e http://192.168.2.18:8096/web/bundle.js?v=10.4.2:25
    pixelBoxStyles http://192.168.2.18:8096/web/bundle.js?v=10.4.2:25
    Ke http://192.168.2.18:8096/web/bundle.js?v=10.4.2:25
    css http://192.168.2.18:8096/web/bundle.js?v=10.4.2:25
    css http://192.168.2.18:8096/web/bundle.js?v=10.4.2:25
    W http://192.168.2.18:8096/web/bundle.js?v=10.4.2:25
    css http://192.168.2.18:8096/web/bundle.js?v=10.4.2:25
    n http://192.168.2.18:8096/web/bundle.js?v=10.4.2:91
    u http://192.168.2.18:8096/web/bundle.js?v=10.4.2:25
    c http://192.168.2.18:8096/web/bundle.js?v=10.4.2:25
Begin ConnectionManager constructor connectionmanager.js:463:17
loading ApiClient singleton site.js:282:29
creating ApiClient singleton site.js:285:33
ApiClient serverAddress: http://192.168.2.18:8096 apiclientcore.js:63:17
ApiClient appName: Jellyfin Web apiclientcore.js:63:75
ApiClient appVersion: 10.4.0 apiclientcore.js:63:121
ApiClient deviceName: Firefox apiclientcore.js:63:173
ApiClient deviceId: TW96aWxsYS81LjAgKFgxMTsgVWJ1bnR1OyBMaW51eCB4ODZfNjQ7IHJ2OjY3LjApIEdlY2tvLzIwMTAwMTAxIEZpcmVmb3gvNjcuMHwxNTU5NjY1NDM2MDA3

...

Requesting url without automatic networking: http://192.168.2.18:8096/Users/81569166e8594d3daba82ceeaddea969 apiclientcore.js:264:21
returning instance from getOrAddApiClient connectionmanager.js:555:21
Requesting url without automatic networking: http://192.168.2.18:8096/DisplayPreferences/usersettings?userId=81569166e8594d3daba82ceeaddea969&client=emby apiclientcore.js:264:21
web socket connection opened apiclientcore.js:142:17
resolving connectToServers with result.State: SignedIn connectionmanager.js:658:29
appRouter - processing path request /web#!/home.html appRouter.js:390:17
appRouter - user is authenticated appRouter.js:411:21
appRouter - proceeding to /web#!/home.html appRouter.js:428:17
RequireJS error: TypeError: "focus" is read-only. Failed modules: site.js:381:17
TypeError: "focus" is read-only
emby-tabs.js:161:5
Requesting url without automatic networking: http://192.168.2.18:8096/System/Endpoint
```
</details>

**Comment by dkanada** [comment]
@grafixeyehero can you confirm I got the dependency versions right? I'm guessing either one of them is wrong or something in the bundle isn't working well on Firefox.

**Comment by grafixeyehero** [comment]
> 
> 
> @grafixeyehero can you confirm I got the dependency versions right? I'm guessing either one of them is wrong or something in the bundle isn't working well on Firefox.

the dependency version was right. i checked in with my previous PR setup against current branch it work well. i will PR for it. something broke in the bundle

**Comment by dmitrylyzo** [comment]
On prettified `bundle.js`, `ShadowDOMPolyfill` cannot appendChild (some test `div` with `position: absolute; left: -11111px; ...`) to the documentElement element. I think that before this commit, `webcomponents` was without `ShadowDOMPolyfill`.
I returned `webcomponents` in bower_components, Firefox started loading.

File: src/components/usersettings/usersettingsbuilder.js [hunk 1]
```diff
@@ -73,10 +70,7 @@ define(['appSettings', 'events'], function (appsettings, events) {
     UserSettings.prototype.get = function (name, enableOnServer) {
         var userId = this.currentUserId;
         if (!userId) {
-            // TODO: I'd like to continue to throw this exception but it causes issues with offline use
-            // Revisit in the future and restore it
-            return null;
-            //throw new Error('userId cannot be null');
+            throw new Error('userId cannot be null');
```
Comment (grafixeyehero) [inline src/components/usersettings/usersettingsbuilder.js]: why?

File: src/components/usersettings/usersettingsbuilder.js [hunk 2]
```diff
@@ -1,25 +1,23 @@
-define(['appSettings', 'events'], function (appsettings, events) {
+define(['appSettings', 'events'], function (appSettings, events) {
     'use strict';
 
     function onSaveTimeout() {
-
-        var self = this;
-        self.saveTimeout = null;
-        self.currentApiClient.updateDisplayPreferences('usersettings', self.displayPrefs, self.currentUserId, 'emby');
+        this.saveTimeout = null;
```
Comment (grafixeyehero) [inline src/components/usersettings/usersettingsbuilder.js]: why? 
`self `is a normal variable and accessible inside the callback. This also has the advantage that you can access the `this `value of the callback itself

File: src/components/usersettings/usersettingsbuilder.js (hunk 1 above)
Comment (dkanada) [reply src/components/usersettings/usersettingsbuilder.js]: We don't have an offline mode and it might be good to actually catch this error.

Comment (grafixeyehero) [reply src/components/usersettings/usersettingsbuilder.js]: We should just remove this entirely.

Comment (dkanada) [reply src/components/usersettings/usersettingsbuilder.js]: Do you mean the error? I think it's right though, the API call will fail without an ID.

Comment (grafixeyehero) [reply src/components/usersettings/usersettingsbuilder.js]: yha

**Pull request merged**